src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java

```diff
@@ -128,8 +128,32 @@ private void executePolicyByProject(String projectId) throws IOException, SQLExc
     List<RetentionRule> datasetRules = retentionRuleDao.findDatasetRulesByProjectId(projectId);
     List<RetentionRule> defaultRules = retentionRuleDao.findDefaultRulesByProjectId(projectId);
     RetentionRule globalDefaultRule = retentionRuleDao.findGlobalRuleByProjectId(projectId);
-    if (datasetRules == null || defaultRules == null || globalDefaultRule == null) {
-      throw new SQLException("Failed to get rules");
+    boolean noDatasetRules = false;
+    boolean noDefaultRules = false;
+    boolean noGlobalRule = false;
+
+    if (datasetRules == null || datasetRules.isEmpty()) {
+      logger.info(
+          String.format("No dataset rules configured within SDRS for project %s.", projectId));
+      noDatasetRules = true;
+    }
+
+    if (defaultRules == null || defaultRules.isEmpty()) {
```

Comment:
Can these multiple if statements be written as if/else or are the condition checks mutually exclusive and require their own ifs?

Reply:
they are independent.